src/components/LoginModal/index.tsx

```diff
@@ -0,0 +1,55 @@
+import Link from 'next/link';
+import React, { useEffect, useRef } from 'react';
+import { IoCloseSharp } from 'react-icons/io5';
+
+import { TINY_API_GOOGLE_LOGIN } from '@/constants/url';
+
+import Button from '../Button';
+import SignInWithGoogleIcon from '../icons/signWithGoogle';
+
+interface LoginModalProps {
+    onClose: () => void;
+    children?: React.ReactNode;
+}
+
+const LoginModal: React.FC<LoginModalProps> = ({ onClose, children }) => {
+    const modalRef = useRef<HTMLDivElement>(null);
+
+    const handleClickOutside = (event: MouseEvent) => {
+        if (modalRef.current && !modalRef.current.contains(event.target as Node)) {
+            onClose();
+        }
+    };
+
+    useEffect(() => {
+        document.addEventListener('mousedown', handleClickOutside);
+
+        return () => {
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
+    }, [onClose]);
+
+    return (
+        <div className="fixed top-0 left-0 w-full h-full flex justify-center items-center bg-gray-800 bg-opacity-50 z-50">
```

Comment:
Nit: you can use dialog tag to define modal

Reply:
Fix in this PR - #53 

[See](https://github.com/Real-Dev-Squad/tiny-site-frontend/pull/53/files#diff-2236e40afeca67fc9fbd0769eef72cc3780f421bf8ff8cbfc8c7f3e2d31c17d3R33-R54)